lib/bundlex/project.ex

```diff
@@ -11,7 +11,17 @@ defmodule Bundlex.Project do
   @src_dir_name "c_src"
   @bundlex_file_name "bundlex.exs"
 
-  @type native_name_t :: atom
+  @type native_name :: atom
+  @type native_interface :: :nif | :cnode | :port
+  @type native_language :: :c | :cpp
+
+  @type os_dep_provider ::
```

Comment:
I think some kind of typedoc would be required to describe how the default values are resolved (in case we don't provide `pkg_configs` or `libs`)